testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/basic/CDIResourceTest.java

```diff
@@ -24,29 +30,58 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
+import java.security.SecurityPermission;
+
+import org.jboss.resteasy.test.cdi.basic.resource.resteasy1082.FooResource;
+import org.jboss.resteasy.test.cdi.basic.resource.resteasy1082.TestApplication;
+import org.jboss.resteasy.test.cdi.basic.resource.resteasy1082.TestServlet;
+
 
 /**
  * @tpSubChapter CDI
  * @tpChapter Integration tests
  * @tpTestCaseDetails Regression test for RESTEASY-1082
  * @tpSince RESTEasy 3.0.16
+ *
+ * Jul 27, 2018 Test rewritten to generated the needed archive and write it to disk.
  */
+
 @RunWith(Arquillian.class)
 @RunAsClient
 public class CDIResourceTest {
 
     protected static final Logger logger = LogManager.getLogger(CDIResourceTest.class.getName());
 
-    static final String fromStr;
+    private static final String WAR_NAME = "RESTEASY-1082.war";
     static final String toStr;
+    static final File exportFile;
 
     static {
-        fromStr = TestUtil.getResourcePath(CDIResourceTest.class, "RESTEASY-1082.war");
         toStr = new StringBuilder()
                 .append(TestUtil.getJbossHome()).append(File.separator)
                 .append("standalone").append(File.separator)
                 .append("deployments").append(File.separator)
-                .append("RESTEASY-1082.war").toString();
+                .append(WAR_NAME).toString();
+        exportFile = new File(FileSystems.getDefault().getPath("target").toFile(), WAR_NAME);
+    }
+
+    @Before
+    public void createArchive() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, WAR_NAME);
+        war.addClasses(FooResource.class,
+                TestApplication.class,
+                TestServlet.class);
+        war.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        war.addAsWebInfResource(CDIResourceTest.class.getPackage(),
+                "web-resteasy1082.xml", "web.xml");
+
+        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+                new SecurityPermission("insertProvider"),
+                new SecurityPermission("insertProvider.BC"),

```

Comment:
These 2 permissions are weird. How is it that this test needs security permissions for the security provider, and BouncyCastle in particular?

Reply:
I mean what makes this different from others in the testsuite, considering it does not even look like a security related test?